examples/example-webhooks-slack/server.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+"""
+This script runs a Flask server that handles webhook requests from Asana,
+and sends a Slack notification whenever the server receives and event indicating
+that a milestone was completed.
+
+Instructions:
+1. Set the ASANA_ACCESS_TOKEN environment variable to your Personal Access Token
+2. Set the SLACK_TOKEN environment variable to your Slack API token
+3. Create a webhook using for your project using manage_webhooks.py
+"""
+
+import hmac
+import os
+import sys
+from hashlib import sha256
+
+import asana
+from flask import Flask, Response, abort, request
+import slack
+
+SECRET_FILE = "/tmp/asana_webhook_secret"
+
+# We need an Asana client to fetch the full names of tasks and users
+# since webhook events are "skinny"
+client = asana.Client.access_token(os.environ.get("ASANA_ACCESS_TOKEN"))
+slack_client = slack.WebClient(token=os.environ.get("SLACK_TOKEN"))
+
+app = Flask(__name__)
+
+
+def load_webhook_secret():
+    with open(SECRET_FILE, "rb+") as f:
+        return f.read().strip()
+
+
+def set_webhook_secret(value):
+    with open(SECRET_FILE, "w+") as f:
+        f.write(value)
+
+
+def validate_request(req):
+    # Takes a flask request and computes a SHA256 HMAC using the webhook
+    # secret we received from Asana and compares it with the signature provided
+    # in the request as described in https://developers.asana.com/docs/#asana-webhooks.
+    # Returns whether the two signatures match.
+
+    request_signature = request.headers.get("X-Hook-Signature")
+    computed_signature = hmac.new(
+        load_webhook_secret(), msg=req.data, digestmod=sha256
+    ).hexdigest()
+
+    return hmac.compare_digest(computed_signature, request_signature)
```

Comment:
👏 since this is optional to get a webhook payload I applaud you including it (I have a feeling a lot of apps don't even check...)